base/frontend/sourcegraph-frontend.Deployment.yaml

```diff
@@ -88,6 +88,33 @@ spec:
         volumeMounts:
         - mountPath: /mnt/cache
           name: cache-ssd
+      - image: sourcegraph/jaeger-agent:1.17.1
+        name: jaeger-agent
+        env:
+          - name: POD_NAME
+            valueFrom:
+              fieldRef:
+                apiVersion: v1
+                fieldPath: metadata.name
+        ports:
+        - containerPort: 5775
+          protocol: UDP
+        - containerPort: 5778
+          protocol: TCP
+        - containerPort: 6831
+          protocol: UDP
+        - containerPort: 6832
+          protocol: UDP
+        resources:
+          limits:
+            cpu: "1"
+            memory: 500M
```

Comment:
is this enough mem for the agent? Wouldn't want to introduce downtime due to the agent getting too many spans.

Is there a way to tell jaeger how much memory to use so it can throw away spans when it is using too much?

Reply:
> is this enough mem for the agent? Wouldn't want to introduce downtime due to the agent getting too many spans.

I believe so. The [default jaeger-agent queue size is 1000 spans](https://www.jaegertracing.io/docs/1.17/performance-tuning/#adjust-server-queue-sizes), after which spans should be dropped in FIFO order. If the jaeger-agent exceeds its memory limit and is OOMKilled, it should be restarted without affecting the availability of the main container. If the jaeger-agent exceeds its memory limit and the pod is evicted, this could cause some downtime while waiting for the new pod to be scheduled, but this is within the "contract" of k8s (i.e., pods might be killed at any point in time, use multiple replicas to ensure reliability).

> Is there a way to tell jaeger how much memory to use so it can throw away spans when it is using too much?

Yes, the jaeger-all-in-one image has a `--memory.max-spans` parameter (which is set in this PR) to limit the maximum number of spans it stores before evicting older spans in FIFO order.